src/context/IssueContext.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+
+import IssuesClient, { Issue } from "@/service/issues";
+
+const IssueContext = createContext<Context | null>(null);
+
+export const IssueProvider = ({children}: Props) => {
+  const [issues, setIssues] = useState<Issue[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const client = new IssuesClient();
+    setIsLoading(true);
+    client.getIssues().then(data => setIssues(data)).catch(setError).finally(() => setIsLoading(false));
+  }, []);
+
+  return <IssueContext.Provider value={{issues, isLoading, error}}>
+    {children}
+  </IssueContext.Provider>
+}
+
+export const useIssue = () => {
+  const issueContext = useContext(IssueContext);
+  if(issueContext) return issueContext;
+  else {
+    throw new Error('잘못된 접근입니다.');
+  }
```

Comment:
위 if 문에서 issueContext 를 반환하기 때문에 else 문 없이 바로 throw 해도 괜찮을거 같아요!

```suggestion
const issueContext = useContext(IssueContext);
  if(issueContext) return issueContext;
  throw new Error('잘못된 접근입니다.');
```